aws/accountid/mode/mode.go

```diff
@@ -0,0 +1,29 @@
+package mode
+
+import "fmt"
+
+// AIDMode switches on/off the account ID based endpoint routing
+type AIDMode string
+
+// enums of valid AIDMode
+const (
+	Preferred AIDMode = "preferred"
+	Required  AIDMode = "required"
+	Disabled  AIDMode = "disabled"
+)
+
+// SetFromString converts config string to corresponding AIDMode or reports error if the value is not enumerated
+func (mode *AIDMode) SetFromString(s string) error {
```

Comment:
There's no real reason to expose this, since sourcing the value from config is an implementation detail for us, let's keep it that way.